base/linalg/triangular.jl

```diff
@@ -1900,14 +1907,10 @@ end
 function sqrtm{T}(A::UnitUpperTriangular{T})
```

Comment:
(Julia doesn't support inheritance of concrete types.)

Indexing expressions `A[i,j]` are slower for `UnitUpperTriangular` than for `UpperTriangular`, and both are slower than for `Array`, because it has to check if `i <= j` and (in the unit case) whether `i == j`.

Since by construction you only access the upper triangle, I would do:
```jl
B = A.data
```
at the beginning of the function (for both the `UnitUpperTriangular` and `UpperTriangular` methods), and then use `B[i,j]` rather than `A[i,j]`.